bundler/lib/bundler/settings.rb

```diff
@@ -428,12 +428,8 @@ def printable_value(value, key)
     def global_config_file
       if ENV["BUNDLE_CONFIG"] && !ENV["BUNDLE_CONFIG"].empty?
         Pathname.new(ENV["BUNDLE_CONFIG"])
-      else
-        begin
-          Bundler.user_bundle_path("config")
-        rescue PermissionError, GenericSystemCallError
-          nil
-        end
+      elsif Bundler.rubygems.user_home && !Bundler.rubygems.user_home.empty?
```

Comment:
@deivid-rodriguez Hey 👋  It seems after this change these env variables are not picked up anymore.

https://github.com/rubygems/rubygems/blob/6a5a1a4af23526e24f21b558517bb8948a6e4ff5/bundler/lib/bundler.rb#L251-L259

Is this expected?

Reply:
Mmmm... right :grimacing:.

I'm guessing you're setting `BUNDLE_USER_CONFIG` to something like `~/.config/bundler` to save the user's bundler configuration to something XDG-compliant, correct?

I broke that, yeah... :disappointed:. I checked these environment variables but I thought they were mostly related caches and missed they also served the change configuration's root (specially since we also have `BUNDLE_CONFIG` to change the configuration location...).

Sorry about that, I'll look into restoring this without reintroducing the original issue.

Reply:
> I'm guessing you're setting BUNDLE_USER_CONFIG to something like ~/.config/bundler to save the user's

That's right. We're using this in CI to setup a bundler config and use `BUNDLE_USER_CONFIG` to point to the config. 

Thanks a lot for looking into this 🙏

Reply:
No problem! In the meantime you can use `BUNDLE_CONFIG` as a workaround. So if you were using `BUNDLE_USER_CONFIG=~/.config/bundler`, you can change it to `BUNDLE_CONFIG=~/.config/bundler/config`.

Reply:
This bit me too. Is there an issue for it yet (I couldn’t find one)? If not, should I create one?

Reply:
Yes, please, because I forgot to create one myself and I had forgotten about it.

Reply:
Okay, thanks, I’ve opened #4793.